connectivity/FEATURE_BLE/include/ble/GattServer.h

```diff
@@ -122,8 +122,9 @@ class GattServer {
         }
 
         /**
-         * Function invoked when the server has sent data to a client as
-         * part of a notification/indication.
+         * Function invoked when the server has sent data to a client. For
+         * notifications this is triggered when data is sent, for indications
+         * it's only triggered when the confirmation has been received.
```

Comment:
So, Cordio executes the "application callback function with confirmation event" whether the client is subscribed to notifications **or** indications, which triggers the call stack that leads to `onDataSent`. The difference is it is triggered in the _sending_ function for notifications but in the _confirming_ function for indications. In other words, [`attsExecCallback`](https://github.com/ARMmbed/mbed-os/blob/331473a70619afbef10832236bd6085bc5cf5f54/connectivity/FEATURE_BLE/libraries/cordio_stack/ble-host/sources/stack/att/atts_ind.c#L145) is called in both cases, which calls [`attExecCallback`](https://github.com/ARMmbed/mbed-os/blob/331473a70619afbef10832236bd6085bc5cf5f54/connectivity/FEATURE_BLE/libraries/cordio_stack/ble-host/sources/stack/att/att_main.c#L377) with [`ATTS_HANDLE_VALUE_CNF`](https://github.com/ARMmbed/mbed-os/blob/331473a70619afbef10832236bd6085bc5cf5f54/connectivity/FEATURE_BLE/libraries/cordio_stack/ble-host/include/att_api.h#L115) as the event, so the Gatt Server cannot differentiate between notifications and indications. 

Perhaps, we could define a separate ATT server callback event for notifications? This way we can call `attExecCallback` directly during the `sending` function, enabling the Gatt Server to differentiate between notifications and indications using the new event type. Of course, this requires modifications to Cordio which might be undesirable.

Reply:
Yeah, I guess you weren't at the standup but that's pretty much my conclusion as well - we could keep the event but then we have to make changes to Cordio and try to upstream them - it's up to Vincent.